src/test/java/trackr/model/task/TaskDeadlineTest.java

```diff
@@ -0,0 +1,46 @@
+package trackr.model.task;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static trackr.testutil.Assert.assertThrows;
+
+import java.time.LocalDate;
+
+import org.junit.jupiter.api.Test;
+
+public class TaskDeadlineTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new TaskDeadline(null));
+    }
+
```

Comment:
Probably should add `constructor_invalidDeadline_throwsIllegalArgumentException()` as well